### ISSUE TITLE
IOError: [Errno 104] Connection reset by peer

### ISSUE THREAD

**Issue by S3phi40T** [sealemar/BingRewards#359]
Hello,

Sometimes I receive IOError: [Errno 104] Connection reset by peer
Recently it happens every day forcing me to perform most searches/clicks manually. 

I bet it's slow network/VPN issue.
I believe this might happen due to timeout for receiving search results? Just wild guess.
If so, can it be tested/increased somehow?

**Comment by aliasfoxkde** [comment]
I found a forum that suggests a try/except block to handle this and explains is clearly:
https://stackoverflow.com/questions/20568216/python-handling-socket-error-errno-104-connection-reset-by-peer

See also:
https://technet.microsoft.com/en-us/library/cc957018.aspx

**Comment by specu** [comment]
just to be clear, try except block catching IOError is there already:
https://github.com/sealemar/BingRewards/blob/master/main.py#L89
so you should  just adjust your count of retries, i can't think of anything else here, but the script catches this instead of crashing.

**Comment by aliasfoxkde** [comment]
@S3phi40T I'm still new to the code, thanks. Troubleshooting methods: can you check your latency, timeout and bandwidth through your connection (it be good to know the conditions of failure). And have you tried pulling down a clean copy of the code. I might be able to replicate your issue later.

**Comment by aliasfoxkde** [comment]
@S3phi40T And the error is being printed but not handled, in your case we are specifically looking for errno.ECONNRESET

**Comment by S3phi40T** [comment]
@aliasfoxkde I'm connecting via VPN and I believe this is the only challenge in here. 
Regards clean code - I'm on @baltimoron release starting with
`git -C $HOME/BingRewards/ pull`
hence I'm always on clean copy :)

Not sure where to look for "errno.ECONNRESET". I'm not a programmer.

**Comment by aliasfoxkde** [comment]
No, it's fine and I like that (auto updating). I'll look later and see if I can handle the error specifically (but first I have to replicate the issue). I like contributing if possible and this is a fun little project.

**Comment by aliasfoxkde** [comment]
The testing conditions proved to be more complicated that I first expected, I did a few tests and wrote the code I believe to resolve the issue but was unable to test it because I need to setup a dummy account for a different region using a slow VPN, and the GUI in my case only shows a fraction of the ones available (OpenVPN through Cli is required). I'm still looking into it, but testing can't be done until this weekend.

**Comment by S3phi40T** [comment]
@aliasfoxkde I'm also on openVPN via CLI.
I have noted everything depends on the latency/profile I use.
Some are better, some are worse.
